### PR TITLE
Add cli function for decoding RNC files.

### PR DESCRIPTION
**Pull request by Alberth289346**
Add main function for decoding RNC files at the cli.

Still needs a CMake wizard, and perhaps some other CI integration wizardry for the finishing touches.

### CODE REVIEW

**Comment by lewri** [comment]
Was this one meant to be a branch inside corsixth rather than your fork? Not that it really matters...

**Comment by TheCycoONE** [comment]
Updated this branch with CMake (and moved some stuff around)

File: CMakeLists.txt [hunk 1]
```diff
@@ -139,6 +140,12 @@ if(BUILD_ANIMVIEW)
   message("Building AnimView")
   add_subdirectory(AnimView)
 endif()
+
+if(BUILD_TOOLS)
+	message("Building Tools")
+	add_subdirectory("tools")
```
Comment (Alberth289346) [inline CMakeLists.txt:146]:
```suggestion
  message("Building Tools")
  add_subdirectory("tools")
```
Replaced TAB by 2 sapces

**Comment by Alberth289346** [comment]
Tried building it. After setting `BUILD_TOOLS`, it generated a working binary. I noticed the option is mentioned in the comments, but it doesn't have a line like
``` cmake
option(ENABLE_UNIT_TESTS "Enables Unit Testing Targets" OFF)
```
No big deal, cli commands are a niche anyway.

Seems fine thus.

**Comment by TheCycoONE** [comment]
> Tried building it. After setting `BUILD_TOOLS`, it generated a working binary. I noticed the option is mentioned in the comments, but it doesn't have a line like
> 
> ```cmake
> option(ENABLE_UNIT_TESTS "Enables Unit Testing Targets" OFF)
> ```
> 
> No big deal, cli commands are a niche anyway.
> 
> Seems fine thus.

Seems more consistent to fix it.

**Pull request merged**